src/codegen/codegen_c_visitor.cpp

```diff
@@ -3481,12 +3481,52 @@ void CodegenCVisitor::print_net_event_call(FunctionCall* node) {
     printer->add_text(")");
 }
 
+/**
+ * Rename arguments to NET_RECEIVE block with corresponding pointer variable
+ *
+ * Arguments to NET_RECEIVE block are packed and passed via weight vector. These
+ * variables need to be replaced with corresponding pointer variable. For example,
+ * if mod file is like
+ *
+ * \code{.mod}
+ *      NET_RECEIVE (weight, R){
+ *          INITIAL {
+ *              R=1
+ *          }
+ *      }
+ * \endcode
+ *
+ * then generated code for initial block should be:
+ *
+ * \code{.cpp}
+ *      double* R = weights + weight_index + 0;
+ *      (*R) = 1.0;
+ * \endcode
+ *
+ * So, the `R` in AST needs to be renamed with `(*R)`.
+ */
+static void rename_net_receive_arguments(ast::NetReceiveBlock* net_receive_node, ast::Node* node) {
+    auto parameters = net_receive_node->get_parameters();
+    for (auto& parameter: parameters) {
+        auto name = parameter->get_node_name();
+        auto var_used = VarUsageVisitor().variable_used(node, name);
+        if (var_used) {
+            RenameVisitor vr(name, "(*" + name + ")");
```

Comment:
```suggestion
            RenameVisitor vr(name, "*" + name);
```
I don't think the `()` are needed.

Reply:
discussed this offline : extra `()` are not needed at the moment but if we will have complex expression involving `[]` or `->`, explicit `()` is useful.